### PR TITLE
Crashdump support for rx65n

### PR DESCRIPTION
**Pull request by RB-tel**
Crashdump spport added to rx65n using standby RAM

### CODE REVIEW

File: arch/renesas/src/rx65n/Kconfig [hunk 1]
```diff
@@ -146,6 +146,11 @@ config RX65N_CMT0
 	bool "CMT0"
 	default y
 
+config RX65N_CMT1
+    bool "CMT1"
+    default n
+    depends on BOARD_CRASHDUMP && RX65N_SBRAM && RX65N_SAVE_CRASHDUMP
```
Comment (patacongo) [inline arch/renesas/src/rx65n/Kconfig:152]: Please use TAB instead of spaces in Kconfig files.

File: arch/renesas/src/rx65n/Kconfig [hunk 2]
```diff
@@ -162,6 +167,15 @@ config RX65N_IRQ_GROUP
...
 
+config RX65N_SBRAM
+     bool "SBRAM"
+     default n
+	 depends on BOARD_CRASHDUMP
+
+config RX65N_SAVE_CRASHDUMP
+     bool "SBRAM Save Crashdump"
+	 depends on RX65N_SBRAM
+
```
Comment (patacongo) [inline arch/renesas/src/rx65n/Kconfig:178]: Please use TAB instead of spaces in Kconfig files.

File: arch/renesas/src/rx65n/Kconfig [hunk 3]
```diff
@@ -288,6 +307,15 @@ config RX65N_IRQ_GROUP
...
```
Comment (patacongo) [inline arch/renesas/src/rx65n/Kconfig:318]: Please use TAB instead of spaces in Kconfig files.

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 4]
```diff
@@ -1,20 +1,36 @@
...
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
  *
```
Comment (patacongo) [inline arch/renesas/src/rx65n/rx65n_definitions.h:34]: No new files can come into the repository without an Apache 2.0 license header.  You can copy the from sched/sched/sched_getcpu.c.

NOTE:  There should be no copyright or author information in the header.

File: arch/renesas/src/rx65n/rx65n_sbram.c [hunk 5]
```diff
@@ -0,0 +1,821 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline arch/renesas/src/rx65n/rx65n_sbram.c:34]: No new files can come into the repository without an Apache 2.0 license header. You can copy the from sched/sched/sched_getcpu.c.

NOTE: There should be no copyright or author information in the header.

File: arch/renesas/src/rx65n/rx65n_sbram.h [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
```
Comment (patacongo) [inline arch/renesas/src/rx65n/rx65n_sbram.h:35]: No new files can come into the repository without an Apache 2.0 license header. You can copy the from sched/sched/sched_getcpu.c.

NOTE: There should be no copyright or author information in the header.

File: boards/renesas/rx65n/rx65n-grrose/src/rx65n_appinit.c [hunk 7]
```diff
@@ -1,20 +1,35 @@
...
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
  *
```
Comment (patacongo) [inline boards/renesas/rx65n/rx65n-grrose/src/rx65n_appinit.c:33]: No new files can come into the repository without an Apache 2.0 license header. You can copy the from sched/sched/sched_getcpu.c.

NOTE: There should be no copyright or author information in the header.

File: boards/renesas/rx65n/rx65n-grrose/src/rx65n_bringup.c [hunk 8]
```diff
@@ -1,20 +1,35 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
  *
  ****************************************************************************/
```
Comment (patacongo) [inline boards/renesas/rx65n/rx65n-grrose/src/rx65n_bringup.c:34]: No new files can come into the repository without an Apache 2.0 license header. You can copy the from sched/sched/sched_getcpu.c.

NOTE: There should be no copyright or author information in the header.

File: boards/renesas/rx65n/rx65n-grrose/src/rx65n_grrose.h [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline boards/renesas/rx65n/rx65n-grrose/src/rx65n_grrose.h:35]: No new files can come into the repository without an Apache 2.0 license header. You can copy the from sched/sched/sched_getcpu.c.

NOTE: There should be no copyright or author information in the header.

File: boards/renesas/rx65n/rx65n-grrose/src/rx65n_sbram.c [hunk 10]
```diff
@@ -0,0 +1,505 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline boards/renesas/rx65n/rx65n-grrose/src/rx65n_sbram.c:34]: No new files can come into the repository without an Apache 2.0 license header. You can copy the from sched/sched/sched_getcpu.c.

NOTE: There should be no copyright or author information in the header.

File: boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_appinit.c [hunk 11]
```diff
@@ -1,20 +1,35 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
  *
  ****************************************************************************/
```
Comment (patacongo) [inline boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_appinit.c:34]: No new files can come into the repository without an Apache 2.0 license header. You can copy the from sched/sched/sched_getcpu.c.

NOTE: There should be no copyright or author information in the header.

File: boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_bringup.c [hunk 12]
```diff
@@ -1,20 +1,35 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
  *
  ****************************************************************************/
```
Comment (patacongo) [inline boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_bringup.c:34]: No new files can come into the repository without an Apache 2.0 license header. You can copy the from sched/sched/sched_getcpu.c.

NOTE: There should be no copyright or author information in the header.

File: boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_rsk2mb.h [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_rsk2mb.h:35]: No new files can come into the repository without an Apache 2.0 license header. You can copy the from sched/sched/sched_getcpu.c.

NOTE: There should be no copyright or author information in the header.

File: boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_sbram.c [hunk 14]
```diff
@@ -0,0 +1,502 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_sbram.c:34]: No new files can come into the repository without an Apache 2.0 license header. You can copy the from sched/sched/sched_getcpu.c.

NOTE: There should be no copyright or author information in the header.

File: arch/renesas/src/rx65n/rx65n_definitions.h (hunk 4 above)
Comment (RB-tel) [reply arch/renesas/src/rx65n/rx65n_definitions.h:34]: @Greg,

With Apache License header, the nxstyle check is failing. The // in the apache url in the license header is creating issue.

Regards,
RB

Comment (patacongo) [reply arch/renesas/src/rx65n/rx65n_definitions.h:34]: > With Apache License header, the nxstyle check is failing. The // in the apache url in the license header is creating issue.

You must be using a very old version of nxstyle.   That was an issue a few months ago.  There are no such complaints from the current version.  There will be no complaints for the PR checking system which uses the latest nxstyle.

File: boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_sbram.c (hunk 14 above)
Comment (RB-tel) [reply boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_sbram.c:19]: Have modified as suggested

File: arch/renesas/src/rx65n/Kconfig [hunk 15]
```diff
@@ -288,6 +307,15 @@ config RX65N_IRQ_GROUP
 	bool "IRQ_GROUP"
 	default y
 
+config RX65N_SBRAM
+     bool "SBRAM"
+     default n
+	 depends on BOARD_CRASHDUMP
+
+config RX65N_SAVE_CRASHDUMP
+     bool "SBRAM Save Crashdump"
```
Comment (patacongo) [inline arch/renesas/src/rx65n/Kconfig:316]: Still need a TAB vs spaces on this line

**Pull request merged**